### PR TITLE
Pairs endpoint implementation

### PR DESCRIPTION
**Pull request by ccorcoveanu**

### CODE REVIEW

File: src/interface.ts [hunk 1]
```diff
@@ -58,6 +59,11 @@ export interface INetworkProvider {
      */
     getNonFungibleTokenOfAccount(address: IAddress, collection: string, nonce: number): Promise<NonFungibleTokenOfAccountOnNetwork>;
 
+    /**
+     * Fetches data about available mex pairs.
+     */
+    getMexPairs(pagination?: IPagination): Promise<PairOnNetwork[]>;
```
Comment (andreibancioiu) [inline src/interface.ts:65]: Perhaps remove this from the interface? If one needs it, one should reference the concrete class, and add interfaces locally.

That is, with respect to:

> The default classes from @elrondnetwork/erdjs-network-providers should only be used as a starting point. As your dApp matures, make sure you switch to using your own network provider, tailored to your requirements (whether deriving from the default ones or writing a new one, from scratch) that directly interacts with the Elrond API (or Gateway).

Extracted from: https://docs.elrond.com/sdk-and-tools/erdjs/extending-erdjs/#extending-the-network-providers

File: src/proxyNetworkProvider.ts [hunk 2]
```diff
@@ -93,6 +94,10 @@ export class ProxyNetworkProvider implements INetworkProvider {
         return tokenData;
     }
 
+    async getMexPairs(_?: IPagination): Promise<PairOnNetwork[]> {
```
Comment (andreibancioiu) [inline src/proxyNetworkProvider.ts:97]: Perhaps remove this?

File: src/interface.ts (hunk 1 above)
Comment (ccorcoveanu) [reply src/interface.ts]: done

File: src/proxyNetworkProvider.ts (hunk 2 above)
Comment (ccorcoveanu) [reply src/proxyNetworkProvider.ts]: done

File: src/proxyNetworkProvider.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@ import { ContractQueryResponse } from "./contractQueryResponse";
 import { DefinitionOfFungibleTokenOnNetwork, DefinitionOfTokenCollectionOnNetwork } from "./tokenDefinitions";
 import { ContractQueryRequest } from "./contractQueryRequest";
 import { EsdtContractAddress } from "./constants";
+import {PairOnNetwork} from "./pairs";
```
Comment (andreibancioiu) [inline src/proxyNetworkProvider.ts]: Unused.

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**